### PR TITLE
Fix missing context for object selection in survey projects

### PR DESCRIPTION
**Pull request by hampelm**

### CODE REVIEW

**Comment by prashtx** [comment]
Is it safer to use `this.listenToOnce(rc, 'sync', function () {...});`? I think there's _probably_ no memory leak with the way you proposed, but I'd have to think about it a little. This way, the context is provided automatically, and the listener is removed as soon as we're done with it.

**Comment by hampelm** [comment]
Good thinking. I'll make that change.

**Comment by hampelm** [comment]
Ok, done & tested

File: src/js/views/projects/datalayers/survey.js [hunk 1]
```diff
@@ -245,14 +245,13 @@ define(function (require) {
...
-      rc.on('sync', function(collection) {
+      this.listenToOnce(rc, 'sync', function(collection) {
         if (collection.length === 0) {
           return;
         }
 
         this.mapView.selectObject(collection.toJSON()[0].geo_info.geometry);
-      });
-
+      }.bind(this));
```
Comment (prashtx) [inline src/js/views/projects/datalayers/survey.js]: With `listenToOnce`, we shouldn't need to bind to `this`.

**Pull request merged**